### ISSUE TITLE
Some USDZ models can't be loaded

### ISSUE THREAD

**Issue by Alyona2209** [ponahoum/three-usdz-loader#6]
For some reason, some USDZ models are not been loaded in the viewer (but they are loaded on iOS). Without special errors, the scene just remains clean:
<img width="1437" alt="Screenshot 2022-09-12 at 11 10 03" src="https://user-images.githubusercontent.com/47988326/189616144-a2a24c69-58d1-45e8-85f5-291862221da1.png">
[usdz (1).zip](https://github.com/ponahoum/three-usdz-loader/files/9546553/usdz.1.zip)
Please, check

**Comment by GregOriol** [comment]
Same here with some usdz files:
```
[Warning] Coding Error: in Usd_GetAPISchemaPluginApplyToInfoForType at line 1527 of /home/schroek/src/USD_schroek/pxr/usd/usd/schemaRegistry.cpp -- Failed to find plugin for schema type 'UsdVolOpenVDBAsset' (emHdBindings.js, line 2970)
[Warning] Coding Error: in Usd_GetAPISchemaPluginApplyToInfoForType at line 1527 of /home/schroek/src/USD_schroek/pxr/usd/usd/schemaRegistry.cpp -- Failed to find plugin for schema type 'UsdVolVolume' (emHdBindings.js, line 2970)
[Warning] Coding Error: in Usd_GetAPISchemaPluginApplyToInfoForType at line 1527 of /home/schroek/src/USD_schroek/pxr/usd/usd/schemaRegistry.cpp -- Failed to find plugin for schema type 'UsdVolField3DAsset' (emHdBindings.js, line 2970)
[Warning] Coding Error: in _GetSchemaKindFromPlugin at line 273 of /home/schroek/src/USD_schroek/pxr/usd/usd/schemaRegistry.cpp -- Failed to find plugin for schema type 'UsdVolOpenVDBAsset' (emHdBindings.js, line 2970)
[Warning] Coding Error: in _GetSchemaKindFromPlugin at line 273 of /home/schroek/src/USD_schroek/pxr/usd/usd/schemaRegistry.cpp -- Failed to find plugin for schema type 'UsdVolVolume' (emHdBindings.js, line 2970)
[Warning] Coding Error: in _GetSchemaKindFromPlugin at line 273 of /home/schroek/src/USD_schroek/pxr/usd/usd/schemaRegistry.cpp -- Failed to find plugin for schema type 'UsdVolField3DAsset' (emHdBindings.js, line 2970)
```

**Comment by LucaFilitz** [comment]
I think this is connected to animations.
See here: #11

**Comment by GregOriol** [comment]
It's most likely on the wasm side of things, but not sure when/how it could be supporting what is missing

**Comment by meshula** [comment]
It seems like the wasm side is not compiled with openvdb support. Maybe your files have simulation data in them?

**Comment by GregOriol** [comment]
@meshula No idea! What about the file provided by @Alyona2209 in the issue?

**Comment by meshula** [comment]
Yeah, my apologies, I just checked the contents of the file. The file is well formed, and the balloon is animated in various programs that I checked. I just opened the file in `usdedit` and searched through it, there's nothing in the file related to the warnings reported coming from `emHdBindings.js`. I misunderstood that you thought they were related to the problem when you meant that you see the same warnings.

**Comment by GregOriol** [comment]
@meshula I'm seeing the errors in the console as I pasted above, which seems to be the same errors as @Alyona2209 saw in the screenshot and with the file attached in the first message of the issue.

Could you maybe also check the files pointed on there https://github.com/ponahoum/three-usdz-loader/issues/11 :
 * https://developer.apple.com/augmented-reality/quick-look/models/stratocaster/fender_stratocaster.usdz => said to not have animations and to work with the loader
 * https://developer.apple.com/augmented-reality/quick-look/models/drummertoy/toy_drummer_idle.usdz => said to have animations and not work with the loader

The files that don't work on my side seem to have animations too. Maybe `toy_drummer_idle.usdz` is a good way to investigate?

**Comment by meshula** [comment]
Those files work in Quicklook and other places. The problem must be with three-usdz-loader itself.

**Comment by ponahoum** [comment]
@meshula @GregOriol @Alyona2209 thanks for the report. The issues you are encountering with those files are very likely to be within the wasm side of things indeed. As mentioned in the readme, this library is just a wrapper around autodesk wasm adaptation of usd. It could be some missing extensions during compilation, or something else. I don’t have much time right now to investigate, but I would encourage you to compile autodesk usd repository yourself, and see if compiling with those extra extensions fixes your issues.

I plan to investigate about all of those issues as soon as I have time, thanks for your understanding !

**Comment by GregOriol** [comment]
After investigations, the issue we were having seems to have nothing to do with those warnings about UsdVol, but with the missing support of UsdSkel as the objects having issues seem to have skeletal animations.

**Comment by GregOriol** [comment]
If someone is interested, I have a working version with UsdSkel support with textures and animations perfectly working on all USDZ files that didn't work before here:
https://github.com/GregOriol/three-usdz-loader/commit/07dfd5c0956d93bfd337a32aa0bba15889de32a3

It has many changes on the autodesk-usd's hdJavaScript branch here:
https://github.com/GregOriol/USD/commit/a58bddb26a26a302214f3b1665bac250f488b293

As well as a build with a newer version of autodesk-usd wasm v25 here:
https://github.com/GregOriol/three-usdz-loader/commit/296dd28502573e742270915086340cc55d3374b3

This one is based on the autodesk-usd's newer adsk/feature/wasm and maintained implementation with many changes:
https://github.com/GregOriol/USD/commit/44f1cedc0af993b5516b29009b5539eaff6d562d

They both run well on the usdz objects we have. The newer implementation is even better.
